starknet/rust/src/lib.rs

```diff
@@ -1,34 +1,35 @@
-use std::ffi::{c_char, CStr, CString};
 use cairo_lang_starknet_classes::casm_contract_class::CasmContractClass;
+use std::ffi::{c_char, CStr, CString};
 
 #[no_mangle]
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
 pub extern "C" fn compileSierraToCasm(sierra_json: *const c_char) -> *mut c_char {
     let sierra_json_str = unsafe { CStr::from_ptr(sierra_json) }.to_str().unwrap();
 
-    let sierra_class =
-        serde_json::from_str(sierra_json_str).map_err(|err| err.to_string());
-    if let Err(e) = sierra_class {
-        return raw_cstr(e)
-    }
+    let sierra_class = match serde_json::from_str(sierra_json_str) {
+        Ok(value) => value,
+        Err(e) => return raw_cstr(e.to_string()),
+    };
+
+    let casm_class = match CasmContractClass::from_contract_class(sierra_class, true, usize::MAX) {
+        Ok(value) => value,
+        Err(e) => return raw_cstr(e.to_string()),
+    };
 
-    let casm_class = CasmContractClass::from_contract_class(sierra_class.unwrap(), true, usize::MAX)
-        .map_err(|err| err.to_string());
-    if let Err(e) = casm_class {
-        return raw_cstr(e)
-    }
+    let casm_json = match serde_json::to_string(&casm_class) {
+        Ok(value) => value,
+        Err(e) => return raw_cstr(e.to_string()),
+    };
 
-    let casm_json = serde_json::to_string(&casm_class.unwrap());
-    if let Err(e) = casm_json {
-        return raw_cstr(e.to_string())
-    }
-    return raw_cstr(casm_json.unwrap())
+    raw_cstr(casm_json)
```

Comment:
Do we really want to return the error msg and the result as indistiguishable return values?

how does the calling function tell the difference. It seems a bit unusal. We could pass in a value to receive the result and return an error if any. 

What do you think @kirugan @AnkushinDaniil 

heres the calling code.
```go
func Compile(sierra *SierraDefinition) (*CompiledClass, error) {
	sierraJSON, err := json.Marshal(SierraDefinition{
		EntryPoints: sierra.EntryPoints,
		Program:     sierra.Program,
		Version:     sierra.Version,
	})
	if err != nil {
		return nil, err
	}

	sierraJSONCstr := C.CString(string(sierraJSON))
	defer func() {
		C.free(unsafe.Pointer(sierraJSONCstr))
	}()

	casmJSONOrErrorCstr := C.compileSierraToCasm(sierraJSONCstr)
	casmJSONOrError := C.GoString(casmJSONOrErrorCstr)
	C.freeCstr(casmJSONOrErrorCstr)

	var casmClass CompiledClass
	if err = json.Unmarshal([]byte(casmJSONOrError), &casmClass); err != nil {
		var syntaxErr *json.SyntaxError
		if errors.As(err, &syntaxErr) {
			return nil, errors.New(casmJSONOrError)
		}
		return nil, err
	}

	return &casmClass, nil
}
```